components/RouteTestDialog.tsx

```diff
@@ -0,0 +1,799 @@
+    });
+
+    return {
+      status: res.status,
+      statusText: res.statusText,
+      headers: resHeaders,
+      body: parsedBody,
+    };
+  } catch (error: unknown) {
+    return {
+      status: 0,
+      statusText: "Error",
+      headers: {},
+      body: (error as Error).message,
+    };
+  }
+}
+
+function prettifyHeader(header: string): string {
+  const replacements: Record<string, string> = {
+    ratelimit: "RateLimit",
+    rpc: "RPC",
+  };
+  return header
+    .split("-")
+    .map((part) => replacements[part] || part.charAt(0).toUpperCase() + part.slice(1))
+    .join("-");
+}
+
+interface RouteTestDialogProps {
+  isOpen: boolean;
+  onClose: () => void;
+  method: RESTMethod;
+  url: string;
+  triggerRef?: RefObject<HTMLElement | null>;
+}
+
+export default function RouteTestDialog({ isOpen, onClose, method, url, triggerRef }: RouteTestDialogProps) {
+  const [pathParams, setPathParams] = useState<Record<string, string>>({});
+  const [queryParams, setQueryParams] = useState<{ key: string; value: string }[]>([{ key: "", value: "" }]);
+  const [optionalQueryParams, setOptionalQueryParams] = useState<string[]>([]);
+  const [token, setToken] = useState("");
+  const [tokenType, setTokenType] = useState<"user" | "bot" | "bearer" | "invalid" | null>(null);
+  const [body, setBody] = useState(method === "GET" ? "" : "{}");
+
+  const [isSettingsOpen, setIsSettingsOpen] = useState(false);
+  const [apiVersion, setApiVersion] = useState("9");
+  const [useCanary, setUseCanary] = useState(false);
+  const [locale, setLocale] = useState("en-US");
+  const [customHeaders, setCustomHeaders] = useState<{ key: string; value: string }[]>([{ key: "", value: "" }]);
+  const [activeTab, setActiveTab] = useState<"body" | "headers">("body"); // Could be expanded to parse rate limits et al
+
+  const [response, setResponse] = useState<{
+    status: number;
+    statusText: string;
+    headers: Record<string, string>;
+    body: unknown;
+  } | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    const matches = url.match(/\{([^}]+)\}/g);
+    if (matches) {
+      const params: Record<string, string> = {};
+      matches.forEach((match) => {
+        const key = match.slice(1, -1);
+        params[key] = "";
+      });
+      setPathParams(params);
+    } else {
+      setPathParams({});
+    }
+    setQueryParams([{ key: "", value: "" }]);
+    setOptionalQueryParams([]);
+    setResponse(null);
+    setBody(method === "GET" ? "" : "{}");
+    setActiveTab("body");
+
+    if (isOpen) {
+      const storedToken = localStorage.getItem("discord_api_token");
+      if (storedToken) {
+        setToken(storedToken);
+        if (BOT_TOKEN_REGEX.test(storedToken)) setTokenType("bot");
+        else if (USER_TOKEN_REGEX.test(storedToken)) setTokenType("user");
+        else if (BEARER_TOKEN_REGEX.test(storedToken)) setTokenType("bearer");
+        else setTokenType("invalid");
```

Comment:
```suggestion
        else setTokenType(null);
```
I wanted to suggest removing this completely but this effect seems to have dependencies.